blog/_src/posts/2017-09-27-tutorial-contributing-to-racket.md

```diff
@@ -0,0 +1,414 @@
+ help via [email](https://groups.google.com/forum/#!forum/racket-users),
+ [IRC](https://botbot.me/freenode/racket/),
+ and/or [Slack](http://racket-slack.herokuapp.com/).
+
+
+## Step 0. Get the Racket source code
+
+The zero-th step in submitting a pull request is getting a recent copy of Racket.
+The simplest way to be up-to-date is to clone Racket from GitHub and build it from source.
+
+```
+$ git clone https://github.com/racket/racket
+$ cd ./racket
+$ make
+```
+<br/>
+
+Running `make` will take some time.
+Once it finishes, you will have a fresh copy of Racket.
+
+```
+$ ./racket/bin/racket --version
+Welcome to Racket v6.10.0.3.
+```
+<br/>
+
+#### Variable Declarations
+
+To ground the discussion for the rest of this post, we need to define a few
+ variables.
+Let:
+
+- `RKT_DIR/` be the directory created (above) by running `git clone`
+- `my-raco` be the executable at `RKT_DIR/racket/bin/raco`
+- `<PKG>` be the name of the package you want to contribute to
+- `my-branch` be any name you feel like using for a Git branch
+
+
+## 1. Fork the package code
+
+The next step is to [fork](https://help.github.com/articles/fork-a-repo/)
+ the code for the package you want to edit and clone the fork to your filesystem.
+The best place to store the clone is in the directory `RKT_DIR/extra-pkgs/`:
+
+```
+$ mkdir RKT_DIR/extra-pkgs
+```
+<br/>
+
+The `extra-pkgs/` directory is the best location because (1) it is close to
+ your development version of Racket and (2) the [`.gitignore`](https://github.com/racket/racket/blob/master/.gitignore#L10)
+ file for the Racket repo ignores this directory.
+But you can use any directory.
+
+To clone a copy of the package `<PKG>`, first try [_raco pkg update_](http://docs.racket-lang.org/pkg/cmdline.html#%28part._raco-pkg-update%29):
+
+```
+$ cd RKT_DIR/extra-pkgs && \
+  my-raco pkg update --clone <PKG>
+```
+<br/>
+
+This command will fail if `<PKG>` is not already installed in your development
+ version of Racket.
+If that happens, use [_raco pkg install_](http://docs.racket-lang.org/pkg/cmdline.html#%28part._raco-pkg-install%29):
+
+```
+$ cd RKT_DIR/extra-pkgs && \
+  my-raco pkg install --clone <PKG>
+```
+<br/>
+
+Lastly, fork the package on GitHub and connect your clone to the fork.
+The high-level steps to follow here are: (1) find the repo on GitHub, click
+ the "fork" button at the top-right; (2) copy the URL for your fork; and (3)
+ use the command-line to add the fork as a remote.
+
+If the URL for your fork is `<FORK_URL>`, the preferred commands to run are:
+
+```
+$ git remote rename origin upstream
+$ git remote add origin <FORK_URL>
+```
+<br/>
+
+These commands give the name `upstream` to the package's GitHub repo
+ and the name `origin` to your fork.
+
+> See also: the GitHub [guide to forking a repo](https://help.github.com/articles/fork-a-repo/).
+
+
+## 2. Find the code
+
+A contribution typically needs an implementation, unit tests, and documentation.
```

Comment:
Consider whether you want to describe a “fast path” for just editing documentation, and then circle back and describe a more involved PR (= code + tests + docs)